sql/cell_value.cc

```diff
@@ -34,9 +34,34 @@ namespace MLDB {
 CellValue::
 CellValue(const Path & path)
 {
+    // Optimized path for when we have just one simple element
+    if (path.size() == 1) {
+
+        const char * p;
+        size_t l;
+        std::tie(p, l) = path.getStringView(0);
+
+        strLength = l;
+        strFlags = 1;  // length of path
+
+        if (strLength <= INTERNAL_LENGTH) {
+            std::copy(p, p + strLength, shortString);
+            type = ST_SHORT_PATH;
+        }
+        else {
+            void * mem = malloc(sizeof(StringRepr) + strLength);
+            longString = new (mem) StringRepr;
+            std::copy(p, p + strLength, longString->repr);
```

Comment:
I added a check of the return value of malloc (duh!) and comments about the noexceptness